src/webhook/test_identity_validation.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+import json
+
+from webhook.identity_validation import get_response
+
+ADMIN_GROUPS = [
+    "osd-sre-admins"
+]
+
+
+def create_request(username, groups, identityName, providerName):
+    class FakeRequest(object):
+        def __init__(self):
+            self.json = {
+                "request": {
+                    "uid": "testuser",
+                    "operation": "UPDATE",
+                    "userInfo": {
+                        "groups": groups,
+                        "username": username,
+                    },
+                    "object": {
+                        "kind": "Identity",
+                        "metadata": {
+                            "name": "identityName"
+                        },
+                        "providerName": providerName
+                    }
+                }
+            }
+
+        def __getitem__(self, item):
+            return getattr(self, item)
+
+    return FakeRequest()
+
+
+class TestIdentityValidation(unittest.TestCase):
+
+    def runtests(self, username, testGroups, identityName, providerName):
+        request = create_request(username, testGroups,
+                                 identityName, providerName)
+
+        response = json.loads(get_response(request, debug=False))
+        return response
+
+    # oauth ServiceAccount can update redhat user identity
+    def test_oauth_sa(self):
+        response = self.runtests(
+            "system:serviceaccount:openshift-authentication:oauth-openshift", "", "OpenShift_SRE:test", "OpenShift_SRE")
+        self.assertTrue(response['response']['allowed'])
+
+    # system:admin can update redhat user identity
+    def test_system_admin(self):
+        response = self.runtests(
+            "system:admin", "", "OpenShift_SRE:test", "OpenShift_SRE")
+        self.assertTrue(response['response']['allowed'])
+
+    # kube:admin can update redhat user identity
+    def test_kube_admin(self):
+        response = self.runtests(
+            "kube:admin", "", "OpenShift_SRE:test", "OpenShift_SRE")
+        self.assertTrue(response['response']['allowed'])
+
+    # users in sre admin groups can update redhat user identity
+    def test_sre_groups(self):
```

Comment:
Recommend add a test to verify SRE can edit a non-SRE identity.

Reply:
done